# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/util-linux/hotfixes.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -ru util-linux-2.11t/lib/nls.h util-linux-2.11t_/lib/nls.h
--- util-linux-2.11t/lib/nls.h	2002-07-04 22:03:26.000000000 +0100
+++ util-linux-2.11t_/lib/nls.h	2002-07-19 12:53:40.000000000 +0100
@@ -1,6 +1,15 @@
 int main(int argc, char *argv[]);
 
 #include "../defines.h"		/* for HAVE_locale_h */
+#ifndef HAVE_locale_h
+#  define HAVE_locale_h 1
+#endif
+
+#ifndef HAVE_langinfo_h
+#  define HAVE_langinfo_h 1
+#endif
+
+
 
 #ifndef PACKAGE
 #define PACKAGE	"util-linux"
diff -ru util-linux-2.11t/MCONFIG util-linux-2.11t_/MCONFIG
--- util-linux-2.11t/MCONFIG	2002-07-04 22:40:07.000000000 +0100
+++ util-linux-2.11t_/MCONFIG	2002-07-19 12:52:47.000000000 +0100
@@ -6,12 +6,14 @@
 #  - remove the `-o root' part in INSTALLSUID
 #  - set USE_TTY_GROUP=no
 #  - define DESTDIR
+# compile the raw utility
+ADD_RAW=yes
 
 # Select for ARCH one of intel, alpha, sparc, arm, m68k, mips
 # Select for CPU i386 if the binaries must be able to run on an intel 386
 # (by default i486 code is generated, see below)
-CPU=$(shell uname -m)
-ARCH=$(shell echo $(CPU) | sed 's/i.86/intel/;s/arm.*/arm/')
+#CPU=$(shell uname -m)
+#ARCH=$(shell echo $(CPU) | sed 's/i.86/intel/;s/arm.*/arm/')
 
 # If HAVE_PAM is set to "yes", then login, chfn, chsh, and newgrp
 # will use PAM for authentication. Additionally, passwd will not be
@@ -115,8 +117,7 @@
   else
     CPUTAIL=486
   endif
-  CPUOPT= $(CPUHEAD)$(CPUTAIL)
-  OPT=          -pipe -O2 $(CPUOPT) -fomit-frame-pointer
+  OPT=          -pipe -O2 -fomit-frame-pointer
 else 
   ifeq "$(ARCH)" "arm"
     OPT=	-pipe -O2 -fsigned-char -fomit-frame-pointer
diff -ru util-linux-2.11t/mount/mount.c util-linux-2.11t_/mount/mount.c
--- util-linux-2.11t/mount/mount.c	2002-07-04 22:29:01.000000000 +0100
+++ util-linux-2.11t_/mount/mount.c	2002-07-19 12:54:26.000000000 +0100
@@ -69,7 +69,8 @@
 #include "env.h"
 #include "nls.h"
 
-#define DO_PS_FIDDLING
+// #define DO_PS_FIDDLING  /* this could be a Problem */
+
 
 #ifdef DO_PS_FIDDLING
 #include "setproctitle.h"
diff -ru util-linux-2.11t/text-utils/Makefile util-linux-2.11t_/text-utils/Makefile
--- util-linux-2.11t/text-utils/Makefile	2002-07-05 21:31:32.000000000 +0100
+++ util-linux-2.11t_/text-utils/Makefile	2002-07-19 12:56:14.000000000 +0100
@@ -50,6 +50,7 @@
 # Do not have ncurses - give up on pg and ul
 pg ul:
 	@echo $@ not made since it requires ncurses
+	@touch $@
 # For more we can also try termcap
 ifeq "$(HAVE_TERMCAP)" "yes"
 more: more.o $(LIB)/xstrncpy.o
@@ -57,6 +58,7 @@
 else
 more:
 	@echo $@ not made since it requires ncurses or termcap
+	@touch $@
 endif
 endif
 
--- ./mount/swap.configure.orig	2002-10-16 12:38:13.000000000 +0200
+++ ./mount/swap.configure	2002-10-16 12:38:36.000000000 +0200
@@ -23,7 +23,7 @@
 echo '#include <unistd.h>
 main(){ exit(0); swapon("/dev/null", 0); }' >> conftest.c
 eval $compile
-if test -s conftest && ./conftest 2>/dev/null; then
+if test -s conftest; then
 	echo "#define SWAPON_HAS_TWO_ARGS" > swapargs.h
 	echo $PAGEH >> swapargs.h
 	echo $SWAPH >> swapargs.h
